tasks/resolve_reference.rb

```diff
@@ -32,77 +35,41 @@ def config_client(opts)
     Aws::EC2::Client.new(options)
   end
 
-  def resolve_reference(opts)
-    client = config_client(opts)
+  def resolve_reference(config, template)
+    unless template.key?(:uri) || template.key?(:name)
+      msg = "You must provide a 'name' or 'uri' in 'target_mapping' for the AWS plugin"
+      raise TaskHelper::Error.new(msg, 'bolt-plugin/validation-error')
+    end
+
+    client = config_client(config)
     resource = Aws::EC2::Resource.new(client: client)
 
     # Retrieve a list of EC2 instances and create a list of targets
     # Note: It doesn't seem possible to filter stubbed responses...
-    resource.instances(filters: opts[:filters]).map do |instance|
+    aws_objects = resource.instances(filters: config[:filters])
+    # This isn't an actual iterable, it's an object with an `each` method
+    targets = []
```

Comment:
The following seems to work for me
```
targets = resource.instances(filters: config[:filters]).select {|i| i.state.name == 'running'}
```
Is that blowing up on automated tests?

Reply:
That's very strange. `resource.instances` returns a `EC2::Instance::Collection` object, which at least according to the docs only has #each, #first, #limit, #size defined... https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EC2/Instance/Collection.html  If select works that's great!

Reply:
Hm I'm really confused. That includes the `Enumerable` ruby module, which is how those methods work. I could swear I tried `map` and `select` both but....I guess not :woman_shrugging:  Good catch Cas! Thanks for trying it out!